deploy.md

```diff
@@ -1,59 +1,142 @@
+At the end of each milestone, we will start a new release process that consists of following tasks.
+
+- creating the `release` branch. A new git branch `release` will be created off the `master` to encapsulate the release preparations, for instance, building release candidates, QA check-ups, and documentation updates etc.
+
+  Once we create this branch and push it to the central repository, the release enters the code-freeze stage, any functionality that isn't already in there is postponed until the next release cycle.
+
+- building release candidate. As soon as the `release` branch lands in the central repo, the CI automatically builds the addon and ships it to the `pre-release` channel, which is supposed to be used by all the stakeholders (QA/PM/Engineers) during the release.
+
+  As a side note, there are three channels for Activity Stream.
+  * [dev](https://s3.amazonaws.com/moz-activity-streams-dev/dist/latest.html)
+  * [pre-release](https://s3.amazonaws.com/moz-activity-streams-prerelease/dist/latest.html)
+  * [release](https://moz-activity-streams.s3.amazonaws.com/dist/latest.html)
+
+- release signing-off. Upon the first build of release candidate, the release engineer sends out the release preview email to all the stakeholders. In addition, she/he files a release issue on Github to track the status through an attached checklist. The release deploy won't be initiated unless all the items get checked by the stakeholders.
+
+  Over the course of this stage, QA tests the pre-release build and files bugs with the label "release/block". Meanwhile, PM and UI/UX verifies the addon and gives feedback. Engineers, in turn, respond to those inputs and make bugfixes or changes accordingly.
+
+- release deploy. As the final stage of the release process, the release engineer ensures all requirements are met on the checklist, then proceeds to deploy the release as described below in detail.
+
+Note that the first three tasks are mostly driven by the automated "pre-release" channel, people from different release party could cooperate directly on Github or via IRC/Slack.
+
+Let's now focus on the final task - release deploy. Again, we break it down into three stages pre-deploy, deploy, and post-deploy respectiviely.
+
+# Pre-deploy
+
+In this stage, we prep everything for the release in the git repo.
+
+## Update code and packages
 
 - ensure you have the latest version of the code
 
-`git pull`
+  ```
+  git checkout release
+  git pull upstream release
+  ```
 
 - update/install any packages
 
-`npm install`
+  ```
+  npm install
+  ```
 
-# Update version numbers
+## Update CHANGELOG and version numbers
 - set a new version of the addon by running npm version
 
-`npm version patch`
-
-- update the changelog:
+  ```
+  npm version patch
+  ```
 
-`./node_modules/.bin/conventional-changelog -i CHANGELOG.md -s -r 0`
+- update the changelog
 
-`git add CHANGELOG.md`
+  ```
+  ./node_modules/.bin/conventional-changelog -i CHANGELOG.md -s -r 0
```

Comment:
Minor aside, but for something this "complex", we may want to convert this into a package.json script so we can just do `npm run changelog`.

Actually, looks like we may already have one, see [package.json:227](https://github.com/mozilla/activity-stream/blob/3c26974142597f9541c089667b551d58436aaf8c/package.json#L227):

```js
"changelog": "conventional-changelog -i CHANGELOG.md -s",
```

So not sure if we want to change this to something like possibly:

```sh
$ npm run changelog -- -r 0
```